src/main/java/com/iexec/core/sms/SmsService.java

```diff
@@ -96,24 +102,22 @@ private boolean checkSmsTeeEnclaveProvider(SmsClient smsClient,
         return true;
     }
 
-    public Optional<String> getEnclaveChallenge(String chainTaskId, boolean isTeeEnabled) {
-        return isTeeEnabled
-                ? generateEnclaveChallenge(chainTaskId)
-                : Optional.of(BytesUtils.EMPTY_ADDRESS);
+    public Optional<String> getEnclaveChallenge(String chainTaskId, String smsUrl) {
+        return StringUtils.isEmpty(smsUrl)
+                ? Optional.of(BytesUtils.EMPTY_ADDRESS)
+                : generateEnclaveChallenge(chainTaskId, smsUrl);
     }
 
     @Retryable(value = FeignException.class)
-    Optional<String> generateEnclaveChallenge(String chainTaskId) {
-        final TaskDescription taskDescription = iexecHubService.getTaskDescription(chainTaskId);
-
+    Optional<String> generateEnclaveChallenge(String chainTaskId, String smsUrl) {
         // SMS client should already have been created once before.
         // If it couldn't be created, then the task would have been aborted.
         // So the following won't throw an exception.
-        final SmsClient smsClient = smsClientProvider.getOrCreateSmsClientForTask(taskDescription);
+        final SmsClient smsClient = smsClientProvider.getSmsClient(smsUrl);
 
         final String teeChallengePublicKey = smsClient.generateTeeChallenge(chainTaskId);
 
-        if (teeChallengePublicKey == null || teeChallengePublicKey.isEmpty()) {
+        if (StringUtils.isEmpty(teeChallengePublicKey)) {
             log.error("An error occurred while getting teeChallengePublicKey [chainTaskId:{}]", chainTaskId);
```

Comment:
Maybe we should only log `smsUrl`, not `chainTaskId`. So we can remove this parameter as it is not used anywhere else. The `chainTaskId` would still be logged in one of the calling methods so we don't lose any info.

Reply:
Let's log sms url too (task id is still required on method). See https://github.com/iExecBlockchainComputing/iexec-core/pull/523#discussion_r977741179 :)